src/discof/restore/utils/fd_sshttp.c

```diff
@@ -149,17 +237,182 @@ fd_sshttp_init( fd_sshttp_t * http,
     }
   }
 
+  if( FD_LIKELY( is_https ) ) {
+    http->state    = FD_SSHTTP_STATE_CONNECT;
+    http->deadline = now + FD_SSHTTP_DEADLINE_NANOS;
+  } else {
+    http->state    = FD_SSHTTP_STATE_REQ;
+    http->deadline = now + FD_SSHTTP_DEADLINE_NANOS;
+  }
+}
+
+#if FD_HAS_OPENSSL
+static int
+http_connect_ssl( fd_sshttp_t * http,
+                  long          now ) {
+  if( FD_UNLIKELY( now>http->deadline ) ) {
+    FD_LOG_WARNING(("deadline exceeded during connect"));
```

Comment:
```suggestion
    FD_LOG_WARNING(( "deadline while connecting to HTTPS snapshot server" ));
```
Might also want to log the peer URL here